networkit/cpp/embedding/BiasedRandomWalk.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * BiasedRandomWalk.cpp
+ *
+ *  Created on: 03.07.2020
+ *      Author: Klaus Ahrens  <ahrens@informatik.hu-berlin.de>
+ *
+ *  adapted and reimplemented from node2vec
+ *  part of snap [https://github.com/snap-stanford/snap]
+ *  Copyright (c) 2007-2019, Jure Leskovec (under BSD license)
+ *
+ *  see [https://arxiv.org/pdf/1607.00653v1.pdf]
+ *
+ */
+
+// networkit-format
+
+#include <algorithm>
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+
+#include <networkit/auxiliary/Random.hpp>
+#include <networkit/graph/Graph.hpp>
+
+#include "AliasSampler.hpp"
+#include "BiasedRandomWalk.hpp"
+
+namespace NetworKit {
+namespace Embedding {
+
+using NeighborSet = std::unordered_set<node>;
+using NeighborMap = std::unordered_map<node, AliasSampler>;
+
+struct GraphData {
+    // assuming contiguous node numbers 0..n-1, they serve as indices
+    using Data = std::vector<NeighborMap>;
+    Data data;
+
+    GraphData(size_t nn) : data{nn, NeighborMap()} {}
+};
+
+std::unique_ptr<GraphData> graphData;
```

Comment:
Avoid to define global variables, store `graphData` as a member variable where needed.